flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/connectors/DynamicSinkUtils.java

```diff
@@ -88,7 +89,8 @@ public static RelNode convertCollectToRel(
             RelNode input,
             CollectModifyOperation collectModifyOperation,
             Configuration configuration,
-            ClassLoader classLoader) {
+            ClassLoader classLoader,
+            ZoneId zoneId) {
```

Comment:
get the ZoneId from configuration? we should reduce the number of arguments to methods esp. if it is encoded in the other arg. Also, we should not pass `TableConfig` anymore as this is a pure API class. `ReadableConfig` should be passed around.